### PR TITLE
feat: Add buy sell credits buttons

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1192

image

add:
- `StickyBar component`
- `CreateSellOrderFlow`
- `SellOrdersActionsBar`

refactor:
- extract common logic to use `CreateSellOrderFlow`and `BuySellOrderFlow` into hooks
- extract `MediaCardImage` component

fix:
- blinking issue with `ProjectMedia` and `MediaCard` component

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-1218--regen-registry.netlify.app/projects/C01-001
2. Try Buy or Sell action

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/cards/MediaCard.Image.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  zIndex: 0,
+  position: 'absolute',
+  bottom: 0,
+  width: '100%',
+  borderRadius: '9px 9px 0px 0px',
+  background:
+    'linear-gradient(180.28deg, rgba(0, 0, 0, 0) 65.91%, rgba(0, 0, 0, 0.6) 99.59%)',
+});
+
+export const MediaCardImage = ({
```
Comment (flagrede) [inline web-components/src/components/cards/MediaCard.Image.tsx:28]: This is just a quick refactor to extract the image from `MediaCard` component.

File: web-components/src/components/image/index.tsx [hunk 2]
```diff
@@ -64,9 +64,11 @@ const Image: React.FC<ImageProps> = ({
   useEffect(() => {
     if (!serverFailed) {
       const clientWidth = rest.width || imgRef?.current?.clientWidth || 0;
-      setWidth(clientWidth);
+      if (clientWidth !== width) {
```
Comment (flagrede) [inline web-components/src/components/image/index.tsx:67]: Prevent render from Image is width is the same.

File: web-registry/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+} from './SellOrdersActionsBar.constants';
+
+type Params = {
+  isSellButtonDisabled: boolean;
+  isBuyButtonDisabled: boolean;
+  onSellButtonClick: () => void;
+  onBuyButtonClick: () => void;
+};
+
+export const SellOrdersActionsBar = ({
```
Comment (flagrede) [inline web-registry/src/components/organisms/SellOrdersActionsBar/SellOrdersActionsBar.tsx:21]: Reusable component display create/buy sell order actions.

**Comment by flagrede** [comment]
@erikalogie You can check the new StickyBar for sell orders here: https://deploy-preview-1218--regen-registry.netlify.app/projects/C01-001

**Comment by erikalogie** [comment]
Great work! A few things:
- Please remove the credit icon from the "sell" button, it should just be on the "buy" button
- If the user is not logged in, then clicking the "sell" button should activate the red banner prompting them to log in with keplr instead of being greyed out with the tooltip.

**Comment by erikalogie** [comment]
Also I just noticed that the buttons move around if I roll over the user profile dropdown: https://monosnap.com/file/dIsyURKyJCSrBIImmWKM2rtBwIL3Y6.

If this can be fixed easily let's do it otherwise it can be a follow up task.

File: web-components/src/components/cards/MediaCard.tsx [hunk 4]
```diff
@@ -73,35 +63,27 @@ export default function MediaCard({
...
-            boxShadow: '0px 0px 4px rgba(0, 0, 0, 0.1)',
-            borderRadius: '50px',
-            textShadow: '0px 0px 2px rgba(0, 0, 0, 0.3)',
-          })}
-        >
-          {tag}
-        </Label>
-      )}
-    </Image>
+  const optimizedImage = useCallback(
```
Comment (flagrede) [inline web-components/src/components/cards/MediaCard.tsx:66]: Prevent image from blinking on each render.

**Comment by flagrede** [comment]
@erikalogie I managed to handle all your feedback:
- credit icon is gone
- buttons are now active when not connected
- also managed to fix the error banner issue you mentioned on another PR. Clicking a button while not connected right after the error banner fade should trigger it again right away. 
- fix the issue with buttons moving while hovering profile dropdown (this was tricky)
- also fix an issue we had with project card images and project details images which were blinking a lot during loading and also when doing an action

**Review by victordelval — APPROVED** [review]

**Comment by victordelval** [comment]
Super!

File: web-registry/src/pages/Marketplace/Storefront/hooks/useResetErrorBanner.tsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ export const useResetErrorBanner = ({
     if (displayErrorBanner) {
       timeout = setTimeout(
         () => setDisplayErrorBanner(false),
-        DEFAULT_DURATION,
+        DEFAULT_DURATION - 1000,
```
Comment (mhagel) [inline web-registry/src/pages/Marketplace/Storefront/hooks/useResetErrorBanner.tsx:22]: Is there a reason for this, or just experimenting?

**Review by mhagel — APPROVED** [review]

File: web-registry/src/pages/Marketplace/Storefront/hooks/useResetErrorBanner.tsx (hunk 5 above)
Comment (flagrede) [reply web-registry/src/pages/Marketplace/Storefront/hooks/useResetErrorBanner.tsx:22]: Yes before we had about that time before being able to show the error banner again once it disappeared.

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**